tests/modules/encoder_base_test.py

```diff
@@ -34,6 +34,26 @@ def setUp(self):
         self.sorting_indices = sorting_indices
         self.restoration_indices = restoration_indices
 
+    def test_non_stateful_states_are_sorted_correctly(self):
+        encoder_base = _EncoderBase(stateful=False)
+        initial_states = (Variable(torch.randn(6, 5, 7)),
```

Comment:
These shapes are confusing me - can you give a comment explaining the shape here?

Reply:
And this is a tuple because it's the hidden state and the memory cell for the LSTM?

Reply:
oh yeah sorry that is weird - it's `num_layers * directions` - thanks pytorch 👍